syntax = "proto3";
package pulse.cdm.bind;
option java_package = "com.kitware.pulse.cdm.bind";
option csharp_namespace = "pulse.cdm.bind";
option optimize_for = SPEED;

import "pulse/cdm/bind/Environment.proto";
import "pulse/cdm/bind/Conditions.proto";

message AnyEnvironmentConditionData
{
  oneof Condition
  {
    InitialEnvironmentalConditionsData  InitialEnvironmentalConditions = 1;
  }
}

message EnvironmentConditionData
{
  ConditionData                         Condition                      = 1;
}

/** @brief The configuration of the environment to use during stabilization. */
message InitialEnvironmentalConditionsData
{
  EnvironmentConditionData              EnvironmentCondition           = 1;
  oneof Option
  {
    EnvironmentalConditionsData         EnvironmentalConditions        = 2;/**<< @brief An environment conditions object with properties to set in the system environmental conditions. */
    string                              EnvironmentalConditionsFile    = 3;/**<< @brief File containing an environment conditions objet with properties to set in the environmental conditions. */
  }
}